File: Capítulo_7/Sets/Método size/size.js
// size(): devolve quantos elementos estão contidos no conjunto. É semelhante à propriedade length de um array.
class Set {
    constructor() {
        this.items = {};
    }

    // Método size
    size() {
        return Object.keys(this.items).length;
    }

    // Outra forma:
    sizeLegacy() {
        let count = 0;
        for (let key in this.items) {
            if (this.items.hasOwnProperty(key)) {
                count++
            }
        }
        return count;
    }
}

File: Capítulo_7/Sets/Método size/Código em Python/size.py
class Set:
    def __init__(self):
        self.items = {}

    # size(): devolve quantos elementos estão contidos no conjunto. É semelhante à propriedade length de um array.
    def size(self):
        return len(self.items)
    
    def size2(self):
        count = 0
        for key in self.items:
            if self.has(key):
                count += 1
        return count
    
